### ISSUE TITLE
[FEA] libcudf is missing a fill / constant-column function.

### ISSUE THREAD

**Issue by eyalroz** [rapidsai/cudf#1075]
libcudf's API should have a function which lets you fill a column with a single, uniform value. This is more useful than it sounds; and on the Python side, RapidsAI has such a function (thus says @kkraus14 ). 

Possible interface:
```
gdf_error fill(
    gdf_column*   column,
    void*         value,
    bool          fill_with_nulls);
```
as the interface suggests, this variant would allow for a "nullable single value".

**Comment by jrhemstad** [comment]
I definitely support moving more functionality from the Python side to the C++ side.

My only suggestion would be to use the `gdf_scalar` struct for `value` instead of just a `void*`. (Note that this `gdf_scalar` is as of yet unmerged, but will be very soon).

https://github.com/rapidsai/cudf/blob/11b67dc96b9a119df4e43f2ccba1e5653df7995d/cpp/include/cudf/types.h#L96

**Comment by eyalroz** [comment]
@jrhemstad : 
1. `gdf_scalar` should obviously be a union (or a variant if this were C++). Why not make it one? You wouldn't need the ugly pointer then.
2. Why is `gdf_scalar` not nullable? Or - do you want a nullptr to mean a null value?

**Comment by jrhemstad** [comment]
> 1. `gdf_scalar` should obviously be a union (or a variant if this were C++). Why not make it one? You wouldn't need the ugly pointer then.

For consistency with `gdf_column`. 

> 2. Why is `gdf_scalar` not nullable? Or - do you want a nullptr to mean a null value?

We couldn't really think of a reason why you'd want a single null value. But representing it as a `nullptr` is one option if necessary.

> 3. Which branch is this type coming from?

https://github.com/rapidsai/cudf/pull/892

**Comment by williamBlazing** [comment]
I think that gdf_scalar should definitely be nullable. There are many situations where something like that is useful. If using nullptr is the easiest way to do it, so be it, having it be more explicit than that seems cleaner to me, but dont care that much.

**Comment by jrhemstad** [comment]
@devavret sounds like the people have spoken. They want `gdf_scalar` to be nullable.

Can you add a `bool is_null` field either in your current PR or as a follow-on PR?

**Comment by devavret** [comment]
😞 follow on PR it is.

**Comment by eyalroz** [comment]
> > 1. `gdf_scalar` should obviously be a union (or a variant if this were C++). Why not make it one? You wouldn't need the ugly pointer then.
> 
> For consistency with `gdf_column`.

Disagree, but I'll go comment there since it doesn't matter for this purpose.

> We couldn't really think of a reason why you'd want a single null value. But representing it as a `nullptr` is one option if necessary.

Same reason why you would want a non-NULL value. I can't think of (almost) any reason I'd take a gdf_scalar if it's non-nullable.

> #892

Expect some comments there!

**Comment by eyalroz** [comment]
So, I want to move forward on this, and leave the question of the use of gdf_scalar for a later point - as it is a matter of style rather of substance, at least for a `fill()` function; and since I need this for the multisearch implementation. @jrhemstad : Is that ok with you?

**Comment by kkraus14** [comment]
Fixed by #1908
